### PR TITLE
Enable TCP states and ThreadPool stats

### PR DESCRIPTION
**Pull request by russcam**
This commit adds the ability to enable TCP and ThreadPool
statistics to be collected when making a request, exposing
these statistics on the IApiCallDetails on the response, as
well as including in the debug information.

Add documentation explaining how to include these statistics
either on a global or per request basis.

### CODE REVIEW

**Comment by russcam** [comment]
Thoughts on including this in 7.9.1, or 7.10.0, @Mpdreamz ?

File: src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs [hunk 1]
```diff
@@ -562,16 +566,10 @@ protected ConnectionConfiguration(IConnectionPool connectionPool, IConnection co
...
-				{
-					var originalCompletedRequestHandler = _completedRequestHandler;
-					var debugCompletedRequestHandler = v ?? (d => Debug.WriteLine(d.DebugInformation));
-					_completedRequestHandler = d =>
-					{
-						originalCompletedRequestHandler?.Invoke(d);
-						debugCompletedRequestHandler.Invoke(d);
-					};
-				});
+					_completedRequestHandler += v ?? (d => Debug.WriteLine(d.DebugInformation)));
```
Comment (Mpdreamz) [inline src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs:572]: 👍 💯

**Review by Mpdreamz — APPROVED** [review]
Change looks good and much welcomed! Definitely can go out in 7.9.1

I wonder if it might be good to create include these types under https://github.com/elastic/elasticsearch-net/tree/master/src/Elasticsearch.Net/Diagnostics and in the Diagnostics namespace?

**Comment by russcam** [comment]
> I wonder if it might be good to create include these types under https://github.com/elastic/elasticsearch-net/tree/master/src/Elasticsearch.Net/Diagnostics and in the Diagnostics namespace?

I was initially thinking of putting them there, then wondered if that should strictly be for `System.Diagnostics` related types. I'll move them there 👍

**Comment by russcam** [comment]
Failures are related to transforms; opened https://github.com/elastic/elasticsearch-net/issues/5004 to address

**Pull request merged**